enhancements/network/egress-qos.md

```diff
@@ -0,0 +1,119 @@
+---
+title: OVN Pods Egress DSCP QoS
+authors:
+  - "@oribon"
+reviewers:
+  - "@trozet"
+  - "@danwinship"
+  - "@tssurya"
+approvers:
+  - TBD
+creation-date: 2022-02-16
+last-updated: 2022-02-23
+status: implementable
+---
+
+# OVN Pods Egress DSCP QoS
+
+## Summary
+
+Not all traffic has the same priority, and when there is contention for bandwidth, there should be a mechanism for objects outside the cluster to prioritize the traffic.
+To enable this, we will use Differentiated Services Code Point (DSCP) which allows us to classify packets by setting a 6-bit field in the IP header, effectively marking the priority of a given packet relative to other packets as "Critical", "High Priority", "Best Effort" and so on.
+
+By introducing a new CRD `EgressQos`, users could specify a DSCP value for packets originating from pods on a given namespace heading to a specified CIDR.
```

Comment:
nit:
```suggestion
By introducing a new CRD `EgressQoS`, users could specify a DSCP value for packets originating from pods on a given namespace heading to a specified CIDR.
```

Reply:
sorry trying to understand this feature, ovnk will mark and the some entity outside the cluster will check DSCP and take some actions so we are talking about service traffic right so CIDR here are ClusterIP or LBServiceIP right ?

Reply:
Yeah some entity outside of the cluster will take actions based on the dscp value, but the CIDRs here are these entities - destinations that are external.
The purpose is to mark egress traffic from specific pods based on the relevant rule in the namespace's EgressQoS: by dstCIDR and dscp value.